### PR TITLE
Refactor kt comment: zod

### PR DESCRIPTION
**Pull request by HansAuger**
Step 1/4:

- [ ] use zod for props schema and typing
- [ ] composition api
- [ ] put css into component
- [ ] other clean ups if necessary

After refactor:
- [ ] Integrate Kotti Translations
- [ ] #253
- [ ] #438

### CODE REVIEW

File: packages/kotti-ui/source/kotti-comment/index.ts [hunk 1]
```diff
@@ -4,20 +4,29 @@ import { attachMeta, makeInstallable } from '../utilities'
...
 
-export const KtComment = attachMeta(makeInstallable(KtCommentVue), META)
-export const KtCommentInput = attachMeta(
-	makeInstallable(KtCommentInputVue),
-	META,
-)
+export const KtComment = attachMeta(makeInstallable(KtCommentVue), {
+	...META,
+	typeScript: {
+		namespace: 'Kotti.Comment',
```
Comment (HansAuger) [inline packages/kotti-ui/source/kotti-comment/index.ts:22]: I think those will add the `TS` labels to the documentation. I can set these to `null` for now and set again when components are refactored into composition api in a future step.

File: packages/kotti-ui/source/kotti-comment/utilities.ts [hunk 2]
```diff
@@ -1,3 +1,4 @@
 import escape from 'lodash/escape'
 
 export const defaultParser = (message: string) => escape(message)
+export const defaultPostEsapeParser = (message: string) => message
```
Comment (FlorianWendelborn) [inline packages/kotti-ui/source/kotti-comment/utilities.ts:4]:
```suggestion
export const defaultPostEscapeParser = (message: string) => message
```

File: packages/kotti-ui/source/kotti-comment/types.ts [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+		isEditable: z.boolean().default(false),
+		message: z.string(),
+		userAvatar: z.string(),
+		userId: z.number(),
+		userName: z.string(),
+	})
+
+	export const propsSchema = commentSchema.extend({
+		dangerouslyOverrideParser: parseFunctionSchema.default(defaultParser),
+		postEscapeParser: parseFunctionSchema.default(defaultPostEsapeParser),
```
Comment (FlorianWendelborn) [inline packages/kotti-ui/source/kotti-comment/types.ts:22]:
```suggestion
		postEscapeParser: parseFunctionSchema.default(defaultPostEscapeParser),
```

File: packages/kotti-ui/source/kotti-comment/types.ts [hunk 4]
```diff
@@ -0,0 +1,41 @@
+import { z } from 'zod'
+
+import { defaultParser, defaultPostEsapeParser } from './utilities'
+
+const commentIdSchema = z.union([z.number(), z.string()])
+export namespace KottiComment {
+	const parseFunctionSchema = z.function().args(z.string()).returns(z.string())
+
+	export const commentSchema = z.object({
+		createdTime: z.string(),
```
Comment (FlorianWendelborn) [inline packages/kotti-ui/source/kotti-comment/types.ts:10]: Are you sure we can make this required? Previously, this wasn’t enforced

File: packages/kotti-ui/source/kotti-comment/types.ts [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+		id: commentIdSchema,
```
Comment (FlorianWendelborn) [inline packages/kotti-ui/source/kotti-comment/types.ts:11]: Are you sure we can make this required? Previously, this wasn’t enforced

File: packages/kotti-ui/source/kotti-comment/types.ts [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+		isDeletable: z.boolean().default(false),
+		isEditable: z.boolean().default(false),
+		message: z.string(),
+		userAvatar: z.string(),
```
Comment (FlorianWendelborn) [inline packages/kotti-ui/source/kotti-comment/types.ts:15]: Are you sure we can make this required? Previously, this wasn’t enforced

File: packages/kotti-ui/source/kotti-comment/types.ts [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+		userId: z.number(),
```
Comment (FlorianWendelborn) [inline packages/kotti-ui/source/kotti-comment/types.ts:16]: Are you sure we can make this required? Previously, this wasn’t enforced

File: packages/kotti-ui/source/kotti-comment/types.ts [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+		userName: z.string(),
```
Comment (FlorianWendelborn) [inline packages/kotti-ui/source/kotti-comment/types.ts:17]: Are you sure we can make this required? Previously, this wasn’t enforced

File: packages/kotti-ui/source/kotti-comment/types.ts (hunk 7 above)
Comment (FlorianWendelborn) [reply packages/kotti-ui/source/kotti-comment/types.ts:16]:
```suggestion
		userId: z.union([z.number(), z.string()]),
```

File: packages/kotti-ui/source/kotti-comment/types.ts [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+	export type PropsInternal = z.output<typeof propsSchema>
+	export type Props = z.input<typeof propsSchema>
+}
+
+export namespace KottiCommentInput {
+	export const propsSchema = z.object({
+		isInline: z.boolean().default(false),
+		parentId: commentIdSchema,
+		placeholder: z.string(),
+		replyToUserId: z.number(),
```
Comment (FlorianWendelborn) [inline packages/kotti-ui/source/kotti-comment/types.ts:35]:
```suggestion
		replyToUserId: z.union([z.number(), z.string()]),
```

File: packages/kotti-ui/source/kotti-comment/types.ts [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+
+	export type PropsInternal = z.output<typeof propsSchema>
+	export type Props = z.input<typeof propsSchema>
+}
+
+export namespace KottiCommentInput {
+	export const propsSchema = z.object({
+		isInline: z.boolean().default(false),
+		parentId: commentIdSchema,
+		placeholder: z.string(),
```
Comment (FlorianWendelborn) [inline packages/kotti-ui/source/kotti-comment/types.ts:34]: Are you sure we can make this required? Previously, this wasn’t enforced

File: packages/kotti-ui/source/kotti-comment/types.ts [hunk 11]
```diff
@@ -0,0 +1,41 @@
...
+		replyToUserId: z.number(),
+		userAvatar: z.string(),
```
Comment (FlorianWendelborn) [inline packages/kotti-ui/source/kotti-comment/types.ts:36]: Are you sure we can make this required? Previously, this wasn’t enforced

File: packages/kotti-ui/source/kotti-comment/types.ts (hunk 9 above)
Comment (FlorianWendelborn) [reply packages/kotti-ui/source/kotti-comment/types.ts:35]: Are you sure we can make this required? Previously, this wasn’t enforced

File: packages/kotti-ui/source/kotti-comment/types.ts [hunk 12]
```diff
@@ -0,0 +1,41 @@
...
+	})
+
+	export type PropsInternal = z.output<typeof propsSchema>
+	export type Props = z.input<typeof propsSchema>
+}
+
+export namespace KottiCommentInput {
+	export const propsSchema = z.object({
+		isInline: z.boolean().default(false),
+		parentId: commentIdSchema,
```
Comment (FlorianWendelborn) [inline packages/kotti-ui/source/kotti-comment/types.ts:33]: Are you sure we can make this required? Previously, this wasn’t enforced

File: packages/kotti-ui/source/kotti-comment/index.ts (hunk 1 above)
Comment (FlorianWendelborn) [reply packages/kotti-ui/source/kotti-comment/index.ts:22]: Yes, it does, but the most important part about the TS label is that it exposes its props as a TS namespace anyway, so I guess it’s fine, especially given that you’re also refactoring the rest of this in the near future

**Review by FlorianWendelborn — APPROVED** [review]

**Pull request merged**